app/src/test/java/org/simple/clinic/overdue/AppointmentCancelReasonTypeAdapterTest.kt

```diff
@@ -0,0 +1,67 @@
+package org.simple.clinic.overdue
+
+import com.google.common.truth.Truth.assertThat
+import junitparams.JUnitParamsRunner
+import junitparams.Parameters
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(JUnitParamsRunner::class)
+class AppointmentCancelReasonTypeAdapterTest {
+
+  @Test
+  @Parameters(method = "reason types to json keys")
+  fun `reason types should be correctly serialized to their json keys`(
+      reason: AppointmentCancelReason,
+      expectedJsonKey: String
+  ) {
+    val serialized = AppointmentCancelReason.TypeAdapter.fromEnum(reason)
+    assertThat(serialized).isEqualTo(expectedJsonKey)
+  }
+
+  @Suppress("unused")
+  fun `reason types to json keys`(): List<List<Any>> {
+    return listOf(
+        listOf(AppointmentCancelReason.PatientNotResponding, "not_responding"),
+        listOf(AppointmentCancelReason.Moved, "moved"),
+        listOf(AppointmentCancelReason.Dead, "dead"),
+        listOf(AppointmentCancelReason.Other, "other")
+    )
+  }
+
+  @Test
+  @Parameters(method = "json keys to reason types")
+  fun `reason types should be correctly deserialized from their json keys`(
+      jsonKey: String,
+      expectedReason: AppointmentCancelReason
+  ) {
+    val deserialized = AppointmentCancelReason.TypeAdapter.toEnum(jsonKey)
+    assertThat(deserialized).isEqualTo(expectedReason)
+  }
+
+  @Suppress("unused")
+  fun `json keys to reason types`(): List<List<Any>> {
+    return listOf(
+        listOf("not_responding", AppointmentCancelReason.PatientNotResponding),
+        listOf("moved", AppointmentCancelReason.Moved),
+        listOf("dead", AppointmentCancelReason.Dead),
+        listOf("other", AppointmentCancelReason.Other)
+    )
+  }
+
+  @Test
+  fun `unknown reason enum should be serialized to its actual value`() {
+    val unknownReason = AppointmentCancelReason.Unknown(actualValue = "abducted_by_joker")
+    val serialized = AppointmentCancelReason.TypeAdapter.fromEnum(unknownReason)
+
+    assertThat(unknownReason.actualValue).isEqualTo(serialized)
```

Comment:
Nitpick: It's usually `assertThat(valueToBeTested).isEqualTo(expectedValue)`. Here, it's the other way round.

Reply:
True, this is a copy pasta mistake. I'll fix this.